src/app/qgisapp.cpp

```diff
@@ -5940,6 +5959,72 @@ QList<QgsMapLayer *> QgisApp::askUserForOGRSublayers( QgsVectorLayer *layer )
   return result;
 }
 
+QList<QgsMapLayer *> QgisApp::askUserForMDALSublayers( QgsMeshLayer *layer )
+{
+  QList< QgsMapLayer * > result;
+  if ( !layer )
+    return result;
+
+  QStringList sublayers = layer->subLayers();
+  if ( sublayers.empty() )
+    return result;
+
+  QgsSublayersDialog chooseSublayersDialog( QgsSublayersDialog::Mdal, QStringLiteral( "mdal" ), this );
+  chooseSublayersDialog.setShowAddToGroupCheckbox( true );
+
+  QgsSublayersDialog::LayerDefinitionList layersDefList;
+  int id = 1;
+  for ( QString layerUri : sublayers )
+  {
+    QgsSublayersDialog::LayerDefinition definition;
+    QStringList parsedUri = layerUri.split( ':' );
```

Comment:
: is not good deliminer, it could be disk drive on windows ? (C:/) are you sure we do not use '::' or something else?

Reply:
It is the delimiter actually used by MDAL :
https://github.com/lutraconsulting/MDAL/blob/3f030ab3c96d46ea7d8c6ab61f5029e7e3c31613/mdal/api/mdal.h#L198

Agree, this will not be well parsed with Windows... Two solutions : 
-first extract the file path that are surely (?) between ""
- change MDAL delimiter

Reply:
Deliminer for multiple SUB meshes is ;; see https://github.com/lutraconsulting/MDAL/blob/3f030ab3c96d46ea7d8c6ab61f5029e7e3c31613/mdal/api/mdal.h#L205

You do not need to split by ":" deliminer anywhere? Just display user whole string in any form that comes back from MDAL_MeshNames and even allow user to modify the string (to for example force different driver)?

Reply:
here, I don't parse to extract the sublayers, but to extract the mesh name to set the mesh layer name in the dialog

Reply:
i understand, I would not parse it and place there whole stuff DRIVER:file:mesh

Reply:
Why not, but that leads to a loooooong name in the tree layers :
![image](https://user-images.githubusercontent.com/7416892/78695323-f27a0a80-78cb-11ea-8614-fba40c480457.png)

Indeed, the name used in the dialog, is also used to give a name to the mesh layer too.
So, maybe, we can still parse the name? And used it only for naming the layer?

Reply:
can you try to load netcdf file as vector layer to how it is there? If we need to parse name, you need to split by ": and :" and not single "